pydata/pydata-sphinx-theme
# Re-align toggle chevrons with right edge of sidebar

Note: this PR is against `feature-focus` branch **not** `main`. Style fixes related to interaction state (focus, hover, current) will be collected in small increments into `feature-focus` before being merged into `main`.

This PR introduces a few code changes:

1. Negative right margin on TOC toggle chevrons so they align with sidebar edge like before. 
2. Modify markup tree when TOC entry is togglable (`.has-children`). Previously: `li > a ~ input ~ label ~ ul`. Now: `li > (div > a ~ label) ~ input ~ ul`. This was done so that the chevron label element would not have to be absolutely positioned.
3. Refactor some of the styles as a consequence of the markup change 

These changes are explained in greater detail in the inline comments.

### Backstory

@jorisvandenbossche left a comment on #1549 showing [chevrons not aligning with the sidebar edge](https://github.com/pydata/pydata-sphinx-theme/pull/1549#issuecomment-1809716127).

This was the result of [removing a `margin-right: -1rem` rule](https://github.com/pydata/pydata-sphinx-theme/pull/1549#discussion_r1383339150) on the `nav.bd-links` element. I removed that rule because it was causing the right edge of the focus rings to get clipped (the sidebar's overflow rules do not permit the part of the focus ring that goes outside the sidebar to be shown). The unintended consequence of removing the negative margin on the container was the chevrons no longer right-aligning with the edge of the sidebar.

### Test plan

1. Go to /examples, use the tab key to move focus through all of the links in the left sidebar. Expand and collapse all of the chevrons with the mouse. Keep checking the look and feel of the focus ring with the chevrons expanded or collapsed. Check the behavior when the focus ring is adjacent to a chevron—above, below, and on the same line.
2. Do the same at /community
3. Check 1 and 2 for both desktop and mobile (sidebar overlay)
